agent/service.go

```diff
@@ -195,14 +196,23 @@ func (as *agentService) Result(ctx context.Context, consumer string) ([]byte, er
 	return as.result, as.runError
 }
 
-func (as *agentService) Attestation(ctx context.Context) ([]byte, error) {
-	// Implement the logic for the Attestation method here
-	// Use the provided ctx parameter as needed
-	var attestation []byte
+func (as *agentService) Attestation(ctx context.Context, reportData []byte, vpml uint) ([]byte, error) {
+	device, err := client.OpenDevice()
```

Comment:
`client.OpenDevice()` is not needed. The method `GetRawQuoteAtLevel` will open the device, fetch the attestation report and close it.